docs/reference/docs/reindex.asciidoc

```diff
@@ -0,0 +1,461 @@
+[[docs-reindex]]
+==== Reindex API
+
+`_reindex`'s most basic form just copies documents from one index to another.
```

Comment:
Both asciidoc files in the PR are exactly as they were in #16861. They will get better when we merge the rest of the PRs that have been merged to master into 2.x. We'll go over them one last time before release (#16618) so maybe we should just skip these?